lib/internal/Magento/Framework/View/DesignLoader.php

```diff
@@ -51,8 +51,10 @@ public function __construct(
     public function load()
     {
         $area = $this->_areaList->getArea($this->appState->getAreaCode());
-        $area->load(\Magento\Framework\App\Area::PART_DESIGN);
-        $area->load(\Magento\Framework\App\Area::PART_TRANSLATE);
-        $area->detectDesign($this->_request);
+        if ($area &&
+            $this->_request &&
+            $this->_request instanceOf RequestInterface) {
+            $area->detectDesign($this->_request);
+        }
     }
```

Comment:
@Vinai @lbajsarowicz 
moved load part to FrontController to reduce frequently load to much